app/src/main/java/com/es/findsoccerplayers/LoginActivity.java

```diff
@@ -0,0 +1,193 @@
+                            Toast.makeText(LoginActivity.this, "Error: " + task.getException().getMessage(), Toast.LENGTH_LONG).show();
+                            prgBar.setVisibility(View.INVISIBLE);
+                        }
+
+                    }
+                });
+            }
+        });
+
+        btnRegister.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(LoginActivity.this, RegisterActivity.class));
+            }
+        });
+
+        btnGoogle.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                signIn();
+            }
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                GoogleSignInAccount account = task.getResult(ApiException.class);
+                firebaseAuthWithGoogle(account.getIdToken());
+                Toast.makeText(LoginActivity.this, "Loggato con successo", Toast.LENGTH_SHORT).show();
+            } catch (ApiException e) {
+                // Google Sign In failed, update UI appropriately
+                Toast.makeText(this, e.toString(), Toast.LENGTH_SHORT).show();
+                prgBar.setVisibility(View.INVISIBLE);
+                // ...
+            }
+        }
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        prgBar.setVisibility(View.INVISIBLE);
+        Log.w("LoginAct","Act Creata");
+    }
+
+    private void firebaseAuthWithGoogle(String idToken) {
+        AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
+        mAuth.signInWithCredential(credential)
+                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        if (task.isSuccessful()) {
+                            // Sign in success, update UI with the signed-in user's information
+                            FirebaseUser user = mAuth.getCurrentUser();
+                            updateUI(user);
+                        } else {
+                            // If sign in fails, display a message to the user.
+                            prgBar.setVisibility(View.INVISIBLE);
+                            Toast.makeText(LoginActivity.this, (CharSequence) task.getException(), Toast.LENGTH_SHORT).show();
+                            updateUI(null);
```

Comment:
qui parte la main activity se il sign in fallisce?

Reply:
si, anche se non riesco a provarla dato che non riesco a far fallire l'autenticazione con google

Reply:
prova a fare accesso con un mail google che non esiste.

Reply:
si ma il login con google consiste nell'accedere all'app tramite gli acocunt google sicronizzati sul telefono, tipo quelli usati per le mail play store ecc, quindi se sono presenti sul telefono non mi può dare errore

Reply:
quindi se il sign in fallisce, si mostra l'errore e la activity corrente rimane questa di login

Reply:
esatto